src/dotnet/commands/dotnet-build/CompileContext.cs

```diff
@@ -54,32 +54,17 @@ public bool Compile(bool incremental)
         {
             CreateOutputDirectories();
 
-            // compile dependencies
-            foreach (var dependency in Sort(_dependencies.ProjectDependenciesWithSources))
-            {
-                if (incremental)
-                {
-                    var dependencyProjectContext = ProjectContext.Create(dependency.Path, dependency.Framework);
-
-                    if (!DependencyNeedsRebuilding(dependencyProjectContext, new ProjectDependenciesFacade(dependencyProjectContext, _args.ConfigValue)))
-                    {
-                        continue;
-                    }
-                }
-
-                if (!InvokeCompileOnDependency(dependency))
-                {
-                    return false;
-                }
-            }
+            return CompileDendencies(incremental) && CompileRootProject(incremental);
```

Comment:
I like this. Though, it starts to feel like we should have an abstraction to do the project compilation, be it root or one of its dependencies. A bunch of the code is duplicated, but applying to a different set of projects. Makes sense?